app/src/androidTest/java/org/simple/clinic/di/TestNetworkModule.kt

```diff
@@ -0,0 +1,23 @@
+package org.simple.clinic.di
+
+import okhttp3.OkHttpClient
+import org.simple.clinic.analytics.NetworkAnalyticsInterceptor
+import org.simple.clinic.di.AppInfoHttpInterceptor
+import org.simple.clinic.di.NetworkModule
+import org.simple.clinic.network.FailAllNetworkCallsInterceptor
+import org.simple.clinic.remoteconfig.ConfigReader
+import org.simple.clinic.user.LoggedInUserHttpInterceptor
+
+class TestNetworkModule : NetworkModule() {
+  override fun okHttpClient(
+      loggedInInterceptor: LoggedInUserHttpInterceptor,
+      appInfoHttpInterceptor: AppInfoHttpInterceptor,
+      networkAnalyticsInterceptor: NetworkAnalyticsInterceptor,
+      configReader: ConfigReader
+  ): OkHttpClient {
+    return super.okHttpClient(loggedInInterceptor, appInfoHttpInterceptor, networkAnalyticsInterceptor, configReader)
+        .newBuilder()
+        .addInterceptor(FailAllNetworkCallsInterceptor)
```

Comment:
Question @vinaysshenoy , why are we using a `FailAllNetworkCallsInterceptor`?

Reply:
There was one specific integration test where we wanted to verify behaviour when the network call fails. Let's review this sometime and see if we can do this in a better way?

Reply:
Actually, I looked at the code right now and I realized that the `shouldFailAll` property is never being set. Which means that the test that was using this interceptor has been removed. I'll remove this.

Reply:
This was removed in `1eaa2f13a301d9a49ffd9906f48e77dc94510ad4`. 😅

Reply:
"Unresolving" this conversation until I remove this class.